x509/codesigning_cert_validations.go

```diff
@@ -145,35 +149,25 @@ func validateCodeSigningKeyUsagePresent(cert *x509.Certificate) error {
 	return nil
 }
 
-func validateCodeSigningExtendedKeyUsage(cert *x509.Certificate, expectedEku x509.ExtKeyUsage) error {
+func validateCodeSigningExtendedKeyUsage(cert *x509.Certificate) error {
 	if len(cert.ExtKeyUsage) == 0 {
 		return nil
 	}
 
 	excludedEkus := []x509.ExtKeyUsage{
-		x509.ExtKeyUsageAny,
```

Comment:
What's the reason of removing `x509.ExtKeyUsageAny`?

Reply:
Because its value is `0`: https://pkg.go.dev/crypto/x509#ExtKeyUsageAny

The original logic is: always pass in `0` as the `expectedEku`, when we find the `expectedEku` in the `certEku`, we set `hasExpectedEku = true` and `continue`. The `continue` skips the `excludedEkus` check which otherwise would fail at `x509.ExtKeyUsageAny`.

Reply:
We have to do this in the original logic because it was designed to be shared by code signing and timestamping, since we have updated it to code signing only, `x509.ExtKeyUsageAny` is a valid `extKeyUsage`.